buildSrc/src/main/kotlin/scripts/variants.gradle.kts

```
@@ -74,31 +74,36 @@ android {
     if (enableSigning) {
         signingConfigs {
             maybeCreate(BuildTypes.RELEASE).apply {
-                storeFile = file(System.getenv("KEYSTORE_FILE_PATH_RELEASE"))
+                val keystorePath = System.getProperty("user.home") + "/work/_temp/keystore/"
+                storeFile = file("$keystorePath/key.keystore")
```

Instead of having to move the keystore to a specific folder, with a specific name, what if we just require a path to the keystore?

```suggestion
                val keystorePath = System.getEnv("WIRE_KEYSTORE_PATH")
                storeFile = file("keystorePath")
```

This way, no need to move to a specific folder, with a specific name.

Just:

```bash
WIRE_KEYSTORE_PATH="path/to/the.keystore"
./gradlew assemble
```

---

When decoding the keystore it's created in the temporary GitHub folder, actually we are using it directly without moving it

---

So we could still do this when running the action:

```shell
WIRE_KEYSTORE_PATH="gh/actions/path/to/the.keystore"
./gradlew assemble
```

The point is that in this current way, the project is mandating where the keystore should be.
We could change it so whoever is running it could tell the project where the keystore is. More flexible, but same complexity, no?

If we ever want to build this outside GH Actions (IIRC F-Droid builds don't run on GH Actions), we'd need to rewrite this again, or we'd need to make sure F-Droid can put the it goes into that same folder, which may or may not be possible.

It's just more flexible for whoever builds it.

---

On a side-note: `val keystorePath = '....'` could be lifted outside `signingConfigs` to avoid repetition 5 times.

```kotlin
   if (enableSigning) {
        val keystorePath = System.getProperty("user.home") + "/work/_temp/keystore/"
        signingConfigs {
            maybeCreate(BuildTypes.RELEASE).apply {
                storeFile = file("$keystorePath/key.keystore")
                storePassword = System.getenv("KEYSTOREPWD_RELEASE")
                keyAlias = System.getenv("KEYSTORE_KEY_NAME_RELEASE")
                keyPassword = System.getenv("KEYPWD_RELEASE")
            }
            maybeCreate(BuildTypes.DEBUG).apply {
                storeFile = file("$keystorePath/key.keystore")
                storePassword = System.getenv("KEYSTOREPWD_DEBUG")
                keyAlias = System.getenv("KEYSTORE_KEY_NAME_DEBUG")
                keyPassword = System.getenv("KEYPWD_DEBUG")
            }
```

And it would be helpful to make sure the keystore exists and throw a meaningful error.

```kotlin
check(File(keystorePath).exists()){
    "Key store not found in path '$keystorePath'. Make sure the path is correct."
}
```